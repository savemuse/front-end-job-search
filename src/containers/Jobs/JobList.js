import React, { Component } from 'react'
import PropTypes from 'prop-types';
import JobItem from './JobItem';

class JobList extends Component {
  componentDidMount() {
    this.props.getJobsRequest();
  }

  render() {
    const { filter_jobs, loading } = this.props;

    if (loading) {
      return (
        <div className="text-align-center empty-container">
          <h3>資料加載中。</h3>
        </div>
      )
    }

    if (filter_jobs.length === 0) {
      return (
        <div className="text-align-center empty-container">
          <h3>您查詢的資料尚無結果，請修改您的查詢條件。</h3>
        </div>
      )
    }

    return (
      <div className="card-container">
        <div className="text-align-center">
          <h3>{`查詢結果共${filter_jobs.length || 0}筆`}</h3>
        </div>
        <ul>
          { filter_jobs.map((job) => (
              <li key={job.id}>
                <JobItem job={job}/>
              </li>
            ))
          }
        </ul>
      </div>
    )
  }
}

JobList.propTypes = {
  loading: PropTypes.bool.isRequired,
  filter_jobs: PropTypes.array.isRequired,
  getJobsRequest: PropTypes.func.isRequired,
}

export default JobList;
