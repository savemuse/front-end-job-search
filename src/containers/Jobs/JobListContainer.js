import { connect } from 'react-redux';
import JobList from './JobList';
import { getJobsRequest } from '../../redux/modules/jobs/action';

const mapStateToProps = state => {
  const { loading, filter_jobs } = state.jobs;
  return {
    loading,
    filter_jobs,
  };
};

const mapDispatchToProps = dispatch => {
  return {
    getJobsRequest: () => {
      dispatch(getJobsRequest());
    },
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(JobList);