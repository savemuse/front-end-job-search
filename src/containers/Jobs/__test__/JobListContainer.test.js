/* eslint-disable */
jest.mock('../JobList');

import React from 'react';
import { Provider } from 'react-redux';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import storeFake from '../../../utils/testUtil';
import JobList from '../JobList';
import JobListContainer from '../JobListContainer';

describe('container <JobList />', () => {
  let subject;
  let component;
  let jobListComponent;

  beforeEach(() => {
    const store = storeFake({
      jobs: {
        loading: false,
        initial_jobs: [],
        filter_jobs: [],
      }
    });
    subject = mount(
      <Provider store={store}>
        <JobListContainer />
      </Provider>
    );

    component = subject.find(JobListContainer);
    jobListComponent = component.find(JobList);
  });

  it('should be called exactly once', () => {
    expect(JobList.mock.calls.length).toBe(1);
  });

  it('should render', () => {
    expect(component.length).toBeTruthy();
    expect(jobListComponent.length).toBeTruthy();
  });

  it('should execute the dispatch function after calling the getJobsRequest function', () => {
    jobListComponent.props().getJobsRequest();
    expect(subject.instance().store.dispatch).toBeCalled();
  });

  it('should be same as snapshot', () => {
    expect(toJson(jobListComponent)).toMatchSnapshot();
  });
});
