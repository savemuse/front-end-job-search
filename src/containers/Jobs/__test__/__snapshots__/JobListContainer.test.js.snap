// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`container <JobList /> should be same as snapshot 1`] = `
<JobList
  filter_jobs={Array []}
  getJobsRequest={[Function]}
  loading={false}
/>
`;
